core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorSplitManager.java

```diff
@@ -24,4 +28,14 @@ default ConnectorSplitSource getSplits(
     {
         throw new UnsupportedOperationException();
     }
+
+    @Experimental(eta = "2023-03-31")
+    default ConnectorSplitSource getSplits(
+            ConnectorTransactionHandle transaction,
+            ConnectorSession session,
+            SchemaFunctionName name,
+            ConnectorTableFunctionHandle function)
+    {
+        throw new UnsupportedOperationException();
```

Comment:
While trying to utilise this great feature I noticed that if this method is not implemented when it should be this exception is caught somewhere internally and end-user sees only something like
`Query 20230117_150200_00027_ibrzg failed: Internal error`
Also there was nothing in server logs apart from:
```
2023-01-17T16:05:23.692+0100	INFO	dispatcher-query-1	io.trino.event.QueryMonitor	TIMELINE: Query 20230117_150520_00028_ibrzg :: FAILED (GENERIC_INTERNAL_ERROR) :: elapsed 2805ms :: planning 2805ms :: waiting 0ms :: scheduling 0ms :: running 0ms :: finishing 0ms :: begin 2023-01-17T16:05:20.886+01:00 :: end 2023-01-17T16:05:23.691+01:00
```

This is not very problematic but having this error propagate correctly would be very helpful to diagnose issues.